src/main/java/mekanism/client/render/transmitter/RenderLogisticalTransporter.java

```diff
@@ -53,81 +62,118 @@ public void render(TileEntityLogisticalTransporter transporter, double x, double
+            if (!itemStack.isEmpty() && itemStack.getItem() instanceof ItemConfigurator) {
+                RayTraceResult pos = mc.player.rayTrace(8.0D, 1.0F);
+                if (pos != null && pos.sideHit != null && pos.getBlockPos().equals(transporter.getPos())) {
                     int mode = ((TileEntityDiversionTransporter) transporter).modes[pos.sideHit.ordinal()];
-
-                    pushTransporter();
-
-                    GL11.glColor4f(1.0F, 1.0F, 1.0F, 0.8F);
-
-                    bindTexture(MekanismRenderer.getBlocksTexture());
-                    GL11.glTranslatef((float) x, (float) y, (float) z);
-                    GL11.glScalef(0.5F, 0.5F, 0.5F);
-                    GL11.glTranslatef(0.5F, 0.5F, 0.5F);
-
-                    int display = getOverlayDisplay(world, pos.sideHit, mode).display;
-                    GL11.glCallList(display);
-
-                    popTransporter();
+                    GlStateManager.pushMatrix();
+                    GlStateManager.enableCull();
+                    GlStateManager.disableLighting();
+                    GlowInfo glowInfo = GLSMHelper.enableGlow();
+                    GlStateManager.shadeModel(GL11.GL_SMOOTH);
+                    GlStateManager.disableAlpha();
+                    GlStateManager.enableBlend();
+                    GlStateManager.blendFunc(SourceFactor.SRC_ALPHA, DestFactor.ONE_MINUS_SRC_ALPHA);
+                    GlStateManager.color(1, 1, 1, 0.8F);
+                    bindTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
+                    GlStateManager.translate((float) x, (float) y, (float) z);
+                    GlStateManager.scale(0.5F, 0.5F, 0.5F);
+                    GlStateManager.translate(0.5F, 0.5F, 0.5F);
+
+                    int display = getOverlayDisplay(pos.sideHit, mode).display;
+                    GlStateManager.callList(display);
+
+                    GLSMHelper.resetColor();
+                    GlStateManager.disableBlend();
+                    GlStateManager.enableAlpha();
+                    GLSMHelper.disableGlow(glowInfo);
+                    GlStateManager.enableLighting();
+                    GlStateManager.disableCull();
+                    GlStateManager.popMatrix();
                 }
             }
         }
+        if (pushed) {
+            //If we did anything we need to pop the matrix we pushed
+            GlStateManager.popMatrix();
+        }
+    }
 
-        GL11.glPopMatrix();
+    /**
+     * Shrink the in transit list as much as possible. Don't try to render things that are in the same spot with the same color
+     */
+    private Collection<TransporterStack> getReducedTransit(Collection<TransporterStack> inTransit) {
+        Collection<TransporterStack> reducedTransit = new ArrayList<>();
+        //TODO: Should this check stack type also. Not sure if it really matters.
+        Set<Integer> progresses = new HashSet<>();
```

Comment:
Working on changing things around a bit so that it is more accurate in its deduplication. (Also going to test if it even still makes a difference about the deduplication after having made the ejector performance improvements. Given rendering one or two extra stacks might be more performant than comparing them all). But marking this as merging things into one set so I don't think a `BitSet` will cut it.